plutus-core/plutus-core/examples/PlutusCore/Examples/Data/List.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications  #-}
+{-# LANGUAGE TypeOperators     #-}
+
+module PlutusCore.Examples.Data.List
+    ( omapList
+    ) where
+
+import           PlutusCore.Core
+import           PlutusCore.Default
+import           PlutusCore.MkPlc
+import           PlutusCore.Name
+import           PlutusCore.Quote
+
+import           PlutusCore.StdLib.Data.Function
+import           PlutusCore.StdLib.Data.List
+
+import           PlutusCore.Examples.Builtins
+
+-- | Monomorphic @map@ over built-in lists.
+--
+-- /\(a :: *) -> \(f : a -> a) ->
+--     fix {list a} {list a} \(rec : list a -> list a) (xs : list a) ->
+--         caseList {a} xs {list a} xs \(x : a) (xs' : list a) -> cons {a} (f x) (rec xs')
+omapList :: Term TyName Name DefaultUni (Either DefaultFun ExtensionFun) ()
+omapList = runQuote $ do
+    a   <- freshTyName "a"
+    f   <- freshName "f"
+    rec <- freshName "rec"
```

Comment:
Totally irrelevant, but GitHub's rendering 'rec' in red.  Does it think it's a keyword or something?

Reply:
It's a keyword indeed, [`rec`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/arrows.html#do-notation-for-commands) to [`ArrowLoop`](https://hackage.haskell.org/package/base-4.15.0.0/docs/Control-Arrow.html#t:ArrowLoop) is what [`mdo`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/recursive_do.html) to [`MonadFix`](https://hackage.haskell.org/package/base-4.15.0.0/docs/Control-Monad-Fix.html#t:MonadFix).

Reply:
I tend to ignore the existence of arrows, though, hence I don't worry about using the behind-an-extension-keyword.